specs/index.rst

```diff
@@ -42,105 +52,145 @@ asset path
 .. note:: All relative paths **MUST** be relative to the folder containing the *assets-manifest* file.
 
 
-File formats
-------------
+An example
+""""""""""
 
-Version 1.0
-"""""""""""
+A pipeline is generating all the CSS and Javascript required for *example.com*. There are four bundles:
 
-An *assets-manifest* file is a JSON file, usually named ``assets-manifest.json``.
-Its toplevel entry **MUST** be a dictionary.
+ * site-wide CSS (logical path ``site.css``)
+ * site-wide Javascript (logical path ``site.js``)
+ * Javascript for a subscription flow, including additional code for communicating with a payment provider (logical path ``subscribe.js``); this is used with ``site.js`` rather than instead of it
+ * CSS for a "lightbox" media browser (logical path ``lightbox.css``); this is used instead of ``site.css``
 
-Example:
+In production, the pipeline will render each bundle as a single asset, including a hash-based fingerprint in the asset path. For instance, the bundle identified by the logical path ``site.css`` might be rendered as an asset with the asset path ``site-4fbcc857.css``; the bundle ``site.js`` rendered as ``site-14ffdec5.js`` and so forth.
 
-.. code-block:: json
+Additionally, each bundle may also have a sourcemap rendered alongside it; the sourcemap for the ``site.js`` bundle would be a bundle with logical path ``site.js.map``, and might result in an asset with asset path ``site-14ffdec5.js.map``
+
+.. warning::
+
+   Sourcemaps **SHOULD** be listed in the manifest, under a logical
+   path created by concatenating ".map" to the end of the logical path
+   for the relevant bundle.
+
+.. note::
+
+   Although it is helpful for any fingerprint used in the asset path
+   of a sourcemap to match the fingerprint of its related rendered
+   asset, this is not always practical depending on how a particular
+   pipeline is configured.
```

Comment:
I had to read this section twice to understand it :/

What about:
```rst
.. note::

    The fingerprint in the asset path of a sourcemap **MAY** match the
    fingerprint of its related rendered asset.
    However, this is not always practical depending on how a particular
    pipeline is configured.
```

Beyond this rephrasing, I'm wondering whether the "use the same fingerprint for sourcemap and rendered asset" is actually a good idea.
Indeed, the rendered asset contains the path of its related sourcemap; this means that the "rendered asset to sourcemap" relation should be handled exactly like the "css to png" relation: hash the sourcemap first, update the path in the rendered asset, and only then hash the rendered asset.

In that case, the note would become:

```rst
.. note::

    For rendered assets referencing a sourcemap,
    the fingerprint of the asset and of the sourcemap **MAY**
    be computed with the algorithm used with dependent assets:
    determine the fingerprint of the sourcemap first, and then compute
    that of the actual asset.
```

What do you think?